src/Raven.Client/Documents/Operations/Replication/ReplicationHubAccess.cs

```diff
@@ -0,0 +1,263 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Json.Serialization;
+using Raven.Client.Util;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.Replication
+{
+    public class ReplicationHubAccess : IDynamicJson
+    {
+        public string Name;
+        public string CertificateBas64;
+        
+        public string[] Incoming;
+        public string[] Outgoing;
+
+        public DynamicJsonValue ToJson()
+        {
+            return new DynamicJsonValue
+            {
+                [nameof(Name)] = Name,
+                [nameof(Incoming)] = Incoming,
+                [nameof(Outgoing)] = Outgoing,
+                [nameof(CertificateBas64)] = CertificateBas64,
+                
+            };
+        }
+
+        public void Validate()
+        {
+            if (string.IsNullOrEmpty(Name))
+                throw new ArgumentNullException(nameof(Name));
+
+            if((Incoming?.Length ?? 0) == 0 && (Outgoing?.Length == 0))
+                throw new InvalidOperationException($"Either {nameof(Outgoing)} or {nameof(Incoming)} must have a value, but both were null or empty");
+
+            ValidateAllowedPaths(Incoming);
+            ValidateAllowedPaths(Outgoing);
+        }
+        
+        
+        void ValidateAllowedPaths(string[] allowedPaths)
+        {
+            if ((allowedPaths?.Length ?? 0) == 0)
+                return;
+
+            foreach (string path in allowedPaths)
+            {
+                if (string.IsNullOrEmpty(path))
+                    throw new InvalidOperationException("Filtered replication AllowedPaths cannot have an empty / null filter");
+
+                if (path[path.Length - 1] != '*')
+                    continue;
+
+                if (path.Length > 1 && path[path.Length - 2] != '/' && path[path.Length - 2] != '-')
```

Comment:
What if someone doesn't follow our default conventions and has docs with customized separators?

Reply:
Then they will have to deal with it. Filtered replication is already going to require some attention from the developer anyway.

Reply:
OK, But maybe instead of hard coding it we can have those separators configurable?

Reply:
Not a good idea, I think. That opens up complexities with different configuration up and downstream.